### PR TITLE
Updates of the cheat sheet for Kokkos 4.4 and 4.5

### PR DESCRIPTION
**Pull request by pzehner**
This PR updates the cheat sheet to the latest version of Kokkos.

### CODE REVIEW

File: utilization.md [hunk 1]
```diff
@@ -157,14 +155,8 @@ Kokkos::View<double*, Kokkos::HostSpace> hostView("hostView", numberOfElements);
...
-// Scratch memory space
-Kokkos::parallel_for(
-    Kokkos::TeamPolicy<>(leagueSize, teamSize),
-    KOKKOS_LAMBDA (const Kokkos::TeamPolicy<>::member_type& team) {
-        // Allocate scratch memory for each team
-        Kokkos::View<double*, Kokkos::ScratchMemorySpace> scratchView(team.team_scratch(1), scratchSize);
-    }
-);
+// Shared host pinned space
+Kokkos::View<double*, Kokkos::SharedHostPinnedSpace> sharedHostPinnedView("sharedHostPinnedView", numberOfElements);
```
Comment (tpadioleau) [inline utilization.md:159]: Why did you remove the scratch space example ?

File: utilization.md [hunk 2]
```diff
@@ -795,13 +787,18 @@ Kokkos::parallel_for(
...
 | `Kokkos::atomic_and(&x, y)`    | `x &= y`             |
-| `Kokkos::atomic_assign(&x, y)` | `x = y`              |
-| `Kokkos::atomic_decrement(&x)` | `x--`                |
-| `Kokkos::atomic_increment(&x)` | `y++`                |
+| `Kokkos::atomic_dec(&x)`       | `x--`                |
+| `Kokkos::atomic_inc(&x)`       | `x++`                |
+| `Kokkos::atomic_lshift(&x, y)` | `x = x << y`         |
 | `Kokkos::atomic_max(&x, y)`    | `x = std::max(x, y)` |
 | `Kokkos::atomic_min(&x, y)`    | `x = std::min(x, y)` |
+| `Kokkos::atomic_mod(&x, y)`    | `x = %= y`           |
```
Comment (tpadioleau) [inline utilization.md:795]:
```suggestion
| `Kokkos::atomic_mod(&x, y)`    | `x %= y`           |
```

**Review by tpadioleau — APPROVED** [review]

File: utilization.md (hunk 1 above)
Comment (pzehner) [reply utilization.md:159]: This is more from Damien's feedback in June than specific to 4.5. From his words, the scratch space is not a "real" memory space, given that it can only be used in very specific situations.

Comment (tpadioleau) [reply utilization.md:159]: Fine for me
